### DISCUSSION TITLE
Next JS app slow when using getServerSideProps

### DISCUSSION THREAD

**Discussion by gmwill934** [vercel/next.js#48205] [Help]
### Summary

Hello

I have a website that needs to fetch some data on the server, hence the need of using `getServerSideProps`.

The call is made to one of our endpoints that is pretty fast (around 200ms), but whenever we use them inside the application, it takes around 4 to 5 seconds to render that data. (happens only once)

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/65828254/230962306-72404a0d-0e38-4950-b6f2-0766377e0afe.png">

Stats from vercel
<img width="491" alt="image" src="https://user-images.githubusercontent.com/65828254/230962509-afc4b587-09b5-418d-becb-50576cf01fbf.png">

Also `lighthouse` indicates that our TBT (total blocking time) is around 5 to 6 seconds.

We have tried:

- remove one dependency that was not being tree shaked
- reducer our bundle by 80%
- installed latest next js version

I know that when using `ssr` the app will have "slower page transitions", but not to this extend. Could I be missing something?

**Comment by icyJoseph** [comment]
This doesn't look normal. GSSP on first navigation is bound to be slower because it makes the call and then generates all of the HTML server side, once that's done it sends all of that to the client.

Could it be that the props you return are too large? That's gonna be included in the `__NEXT__DATA__` script tag that contains the page's initial setup, and surely would make the parsing stage take very long.
